mantidimaging/core/operations/loader.py

```diff
@@ -2,19 +2,31 @@
 # SPDX - License - Identifier: GPL-3.0-or-later
 import os
 import pkgutil
-from typing import List, Protocol, cast
+import sys
+from typing import List, Protocol, cast, Union, TYPE_CHECKING
 from importlib.machinery import FileFinder, ModuleSpec
 from importlib.abc import Loader
 
+if TYPE_CHECKING:
+    from PyInstaller.loader.pyimod02_importers import FrozenImporter
+
 from mantidimaging.core.operations.base_filter import BaseFilter
 
-MODULES_OPERATIONS = {}
-for finder, module_name, _ in pkgutil.walk_packages([os.path.dirname(__file__)]):
-    assert isinstance(finder, FileFinder)
-    spec = finder.find_spec(module_name)
-    assert isinstance(spec, ModuleSpec)
-    assert isinstance(spec.loader, Loader)
-    MODULES_OPERATIONS[module_name] = spec.loader
+MODULES_OPERATIONS: dict[str, Union['FrozenImporter', Loader]] = {}
+if not MODULES_OPERATIONS:
+    for finder, module_name, _ in pkgutil.walk_packages([os.path.dirname(__file__)]):
+        if getattr(sys, 'frozen', False):
```

Comment:
The `frozen` attribute is only present if the application has been packaged. The PyInstaller docs recommend checking this way (https://pyinstaller.org/en/stable/runtime-information.html) and there's a discussion [here](https://stackoverflow.com/questions/59238237/why-use-getattr-instead-of-hasattr-for-sys-frozen) about why `getattr` is used instead of `hasattr`.